### PR TITLE
Arguments Handling

### PR DESCRIPTION
**Pull request by 31Nathan**
ArgumentsParser Class Created (inherited from AParser)
The Arguments are stored in the class, waiting to be used for maybe a Plazza Class
If the parsing fails, it only prints a message on standard output (waiting for logger)

### CODE REVIEW

File: src/Parser/ArgumentsParser/ArgumentsParser.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+** EPITECH PROJECT, 2022
+** PLAZAAAAAAAAAAAAA
+** File description:
+** ArgumentsParser
+*/
+
```
Comment (Cavonstavant) [inline src/Parser/ArgumentsParser/ArgumentsParser.hpp:7]: `/// \file src/Parser/ArgumentsParser/ArgumentsParser.hpp`

File: src/Parser/ArgumentsParser/ArgumentsParser.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#ifndef ARGUMENTSPARSER_HPP_
+#define ARGUMENTSPARSER_HPP_
+
+#include "../AParser.hpp"
```
Comment (Cavonstavant) [inline src/Parser/ArgumentsParser/ArgumentsParser.hpp:11]: Relative include path will be removed in favor of a simpler `#include "AParser.hpp"`

File: src/Parser/ArgumentsParser/ArgumentsParser.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                    setRefillTime(value);
+                    break;
+                case 1:
+                    setCookNumber(value);
+                    break;
+                case 0:
+                    setCookingTime(value);
+                    break;
+                default:
+                    std::cout << "Problem" << std::endl;
```
Comment (osvegn) [inline src/Parser/ArgumentsParser/ArgumentsParser.cpp:40]: Which type of problem ?

File: src/Parser/CMakeLists.txt [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
 set (SRC_ROOT ${PROJECT_SOURCE_DIR}/src/Parser)
+set (ARG_ROOT ${SRC_ROOT}/ArgumentsParser)
 
 set (SRC
     ${SRC_ROOT}/IParser.hpp
     ${SRC_ROOT}/AParser.hpp
     ${SRC_ROOT}/AParser.cpp
+    ${ARG_ROOT}/ArgumentsParser.hpp
+    ${ARG_ROOT}/ArgumentsParser.cpp
 )
```
Comment (Cavonstavant) [inline src/Parser/CMakeLists.txt:12]: Don't forget to subdirectory to the target include directory using:
`target_include_directory(${PROJECT_NAME}_parser PUBLIC ${ARG_ROOT})`

File: src/Parser/ArgumentsParser/ArgumentsParser.cpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                    break;// raise
+            }
+        } catch (...) {
+            std::cout << "Arg isn't a int" << std::endl;
+            return;// raise
+        }
```
Comment (osvegn) [inline src/Parser/ArgumentsParser/ArgumentsParser.cpp:46]: Usefull to try the switch part ?

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
Need minor fixes and compilation unit update, everything ok otherwise 
image

**Review by osvegn — APPROVED** [review]
LGTM

**Review by osvegn — APPROVED** [review]
LGTM

**Review by Cavonstavant — APPROVED** [review]
LGTM!

**Pull request merged**